src/mon/MonClient.cc

```diff
@@ -960,6 +960,11 @@ void MonClient::tick()
       schedule_tick();
     });
 
+  if (!_opened()) {
```

Comment:
@rzarzynski The only path i hit that situation is during Authorization failure (EAGAIN), then we will be in that state of no active_con and not hunting, all other failures are handled (during hunting)

if we will not check that during the tick, we won't even try other functions that will do the re-open session

Reply:
I'm not sure we want to force a connection in this fashion. If the only way we end up in this state is an authentication failure, shouldn't we be triggering a reopen based on that error, instead of whenever we happen to tick()?

Reply:
@gregsfortytwo not exactly forcing reconnect on every tick, only if not _open which check that there is no active_connection or not hunting

Reply:
I think the worry is that the current solution is a bit like a hammer – very universal, able to even more paths than we're aware about at the moment. On the other hand, this might be even advantageous. Now we have the `if (!_open()) { _reopen_session(); }` idiom in multiple places and still not handling everything.

Reply:
@gregsfortytwo @rzarzynski I added another commit to that PR, it will make tick bit more readable since we already have a connection. 
I just wanted to add one more clarification, in that case that we are not hunting and there is no active_con we will not send any beacon, we will lose the monclient (since tick won't do any reconnect) and MON will remove the mgr.

Reply:
@gregsfortytwo @rzarzynski can you please take another look here?

Reply:
I'm pretty sure that as written this adds a race with ```shutdown()```.

If we want the big hammer (maybe — I'm not convinced, because it's a big hammer and those can obscure deliberate design choices, but somebody else could dig into MonClient enough to be sure) then it certainly needs to make sure we don't have that kind of problem. (I'm not telling you the race because anybody who wants to make this kind of change needs to be informed enough to find it, along with the more-concerning potential for other issues I haven't identified.)